pipelines/nextflow/modules/genome_metadata/datasets_metadata.nf

```diff
@@ -0,0 +1,40 @@
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+process DATASETS_METADATA {
+    tag "$accession"
+    label 'local'
+    label 'cached'
+
+    input:
+        val(accession)
+
+    output:
+        tuple val(accession), path("ncbi_meta.json")
+        
+    shell:
+        '''
+        datasets summary genome accession !{accession} | jq '.' > ncbi_meta.json
+        if [ ! -s "ncbi_meta.json" ]; then
+            echo "No Metadata from datasets for !{accession}"
+            exit 1
+        fi
```

Comment:
There are actually 2 types of "errors" from `datasets`: badly formatted accession (prints message in stderr) or accession not matching anything in NCBI (`{"total_counts": 0}`). I believe we want to consider both:
```suggestion
        datasets summary genome accession !{accession} > ncbi_meta.json
        if [ "$?" -ne 0 ]; then
            echo "Invalid or unsupported assembly accession: !{accession}"
            exit 1
        elif [[ $(jq -r '.total_count' ncbi_meta.json) -eq 0 ]]; then
            echo "No metadata returned for !{accession}"
            exit 1
        fi
```
Note that I have removed `| jq '.'` because if left it "nullifies" `$?` (makes it 0 whatever `datasets` returned).